canonical-web-and-design/vanilla-framework
# Theming spike

## Done

Introduces theming infrastructure and themes hr, nav, p-search-box. Note the subnav not themed in this spike.

- theming settings file _settings_theming.scss
- theming color maps in _settings_colors
- separates colors from rest of rules

Drive by fixes:

1. Removes button styling applied using the [type='submit'] selector, as it has very high specificity and is difficult to override. Example: green border  on hover on the current ubuntu.com search in nav:
![image](https://user-images.githubusercontent.com/2741678/63439090-55925c00-c425-11e9-9afa-18c030da9f0c.png)

2. p-search-box improvements:
 - focus states of button, reset, and p-search-box (the outer container) no longer overlap, and are same height
 - norder of magnifier button removed after consultation with @kwm14 
 - margin-bottom matches margin-bottom on select, other inputs, etc.
 - makes input absolutely positioned, while the reset and search buttons become relatively positioned on top. This allows removal of a lot of hacks and hardcoded vsalues around the positioning of those elements

3. Button improvements:
- Realigns horizontal padding on buttons with other components using $sph-inter With 1.7, left/right button padding was aligned with other elements using the $sph-inter variable. At some point later this changed to $sph-inter--small * 1.5, which doesn't match anything else.
- Improves icons inside buttons - these used display: inline-block, and hacks like top: 2px to counteract the extra white space inline-block causes. This has been replaced with flex, which behaves more predictably and doesn't require hacks. It should be thoroughly tested though.

Also fixes #2449

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Open /examples/patterns/search-box/navigation/ and QA the example.
The default is currently set to light. Inspect on all screen sizes; add is-dark to both nav and search box. Type something into the searchbox, press tab 3 times to see focus outline not overlapping between input, reset, button
- Change the value for keys nav, p-search-box in _settings_themes.scss to true. Repeat the above, adding is-white instead of is-dark to see the alternative to the default theme.

Inspect hr example too.

Documentation forthcoming.

## Details

[List of links to issues/bugs and cards if needed]

## Screenshots

[if relevant, include a screenshot]
